src/event/mod.rs

```diff
@@ -522,14 +527,7 @@ impl<'a> Iterator for FieldsIter<'a> {
     type Item = (&'a Atom, &'a Value);
 
     fn next(&mut self) -> Option<Self::Item> {
-        loop {
-            let (key, value) = match self.inner.next() {
-                Some(next) => next,
-                None => return None,
-            };
-
-            return Some((key, &value));
-        }
+        self.inner.next().map(|(key, value)| (key, value))
```

Comment:
I think you can actually get rid of the `map` altogether.

Reply:
Woops, you're totally right